### PR TITLE
Bluetooth: host: resume advertising even when BT_EXT_ADV=y

### LINKED ISSUE #53048: Bluetooth: legacy advertising doesn't resume if CONFIG_BT_EXT_ADV=y
**Describe the bug**
In this configuration: HCI over RPMSG, `CONFIG_BT_EXT_ADV=y`, `CONFIG_BT_MAX_CONN>1`:
- a legacy advertiser is started using `bt_le_adv_start`, _without setting_ `BT_LE_ADV_OPT_ONE_TIME`.
- a device connects
- `bt_le_adv_resume` fails to resume the advertiser

See the original discussion in https://github.com/zephyrproject-rtos/zephyr/pull/51309 .

edit: wasn't really related to nRF53, also reproduced with just a single device with `CONFIG_BT_EXT_ADV=y`.

**To Reproduce**
- flash `samples/bluetooth/peripheral` with the following config: `CONFIG_BT_MAX_CONN=2`, `CONFIG_BT_EXT_ADV=y` to the nRF5340 app core.
- flash `samples/bluetooth/hci_rpmsg` with the following config: `CONFIG_BT_MAX_CONN=2`, `CONFIG_BT_EXT_ADV=y` to the nRF5340 net core.

**Expected behavior**
Advertising resumes after a connection is established.

### PR DESCRIPTION
**Pull request by jori-nordic**
When using extended advertising commands, the advertising set is not considered to be over until we receive `BT_HCI_EVT_LE_ADV_SET_TERMINATED` from the controller. Only then do we clear the `BT_ADV_ENABLED` flag.

The problem is that `bt_le_adv_resume` is called on connection established, which can happen before adv set terminated, and in that case it will early-return because `BT_ADV_ENABLED` is still set.

This change triggers `bt_le_adv_resume` when we get `BT_HCI_EVT_LE_ADV_SET_TERMINATED`.

Fixes #53048

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

### CODE REVIEW

**Comment by jori-nordic** [comment]
adding DNM until I fix the test I'm extending (ref https://discord.com/channels/720317445772017664/1014241011989487716/1068520429670641825)

**Comment by jori-nordic** [comment]
> adding DNM until I fix the test I'm extending (ref https://discord.com/channels/720317445772017664/1014241011989487716/1068520429670641825)
> 
> stack changes can still be reviewed.

will be monitored by @aescolar since it doesn't seem to be an issue in the test itself

**Review by alwa-nordic — DISMISSED** [review]

**Comment by jori-nordic** [comment]
@hermabe , @jhedberg would any of you have the time to do a quick pass on this?

File: tests/bluetooth/bsim_bt/bsim_test_adv_resume/test_scripts/_compile.sh [hunk 1]
```diff
@@ -9,5 +9,8 @@ bash_source_dir="$(realpath "$(dirname "${BASH_SOURCE[0]}")")"
 source <("${bash_source_dir}/_env.sh")
 
 # Place yourself in the test's root (i.e. ./../)
-west build -b nrf52_bsim && \
-    cp build/zephyr/zephyr.exe "${peripheral_exe}"
+# west build -b nrf52_bsim -d build_test && \
+#     cp build_test/zephyr/zephyr.exe "${test_exe}"
```
Comment (hermabe) [inline tests/bluetooth/bsim_bt/bsim_test_adv_resume/test_scripts/_compile.sh:13]: Nit: No need for this comment.

**Review by hermabe — DISMISSED** [review]
Looks reasonable

File: tests/bluetooth/bsim_bt/bsim_test_adv_resume/test_scripts/_env.sh [hunk 2]
```diff
@@ -11,8 +11,8 @@ bsim_bin="${BSIM_OUT_PATH}/bin"
 verbosity_level=2
```
Comment (aescolar) [inline tests/bluetooth/bsim_bt/bsim_test_adv_resume/test_scripts/_env.sh:11]: I know this PR is not the one introducing this script, and as such this comment is a bit off topic, but 2 things about the script:
* It is very weird to have a script be executed, having it printing out to stdout commands and then have those commands be interpreted by the user script. You can just define any variables you want in a script and have the user simply do `source my_script.sh`. Meaning this script does not need the print_var function or its calls.
* Minor: This script aim seems to be to share the variable names and directory for the test executable. But you are also setting other variables (sim_id, verbosity) here used in only one place (which hides them, and makes the code more complex to understand). I wouldn't define here neither of the later 2.

**Review by aescolar — COMMENTED** [review]
Just a few minor things which I would recommend fixing.

File: tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/bs_bt_utils.c [hunk 3]
```diff
@@ -14,6 +14,9 @@ BUILD_ASSERT(CONFIG_BT_ID_MAX == 2, "CONFIG_BT_ID_MAX should be 2.");
 #define BS_SECONDS(dur_sec)    ((bs_time_t)dur_sec * USEC_PER_SEC)
 #define TEST_TIMEOUT_SIMULATED BS_SECONDS(60)
 
+DEFINE_FLAG(flag_is_connected);
```
[the remark this answers is not in the document]
Comment (Thalley) [reply tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/bs_bt_utils.c:17]: To handle multithreading just doing `bool flag_is_connected` isn't as good IMO. We could make it as `volatile`, but the idea of using `atomic_t` for global variables that multiple threads read/write to (or semaphores) is better IMO. 

The idea of defining atomic flags like this, is something we have in multiple BSIM tests (originating from the LE Audio tests).

File: tests/bluetooth/bsim_bt/bsim_test_adv_resume/test_scripts/_env.sh (hunk 2 above)
Comment (jori-nordic) [reply tests/bluetooth/bsim_bt/bsim_test_adv_resume/test_scripts/_env.sh:11]: Yeah I find this script weird too, I think @alwa-nordic introduced it, he was using it for local development. I'll simplify it.

File: tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/bs_bt_utils.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (aescolar) [reply tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/bs_bt_utils.c:17]: Up to you guys. This code will do the same with this atomic type and using the atomic API or without.

**Review by aescolar — APPROVED** [review]
I'm ok on a quick look. For the host change I have no opinion though.

**Review by alwa-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**